### DISCUSSION TITLE
How to gracefully handle and exit when sheet is empty

### DISCUSSION THREAD

**Discussion by rbharathkumar** [nightscape/spark-excel#711] [Q&A]
Hi All,
Hope you are all doing well.
With your help a long time ago, I am successfully able to read an excel with multiple sheets.
Sheets are read as suggested by getting the sheet names in advance and looping through them.
Also read with schema and has a header.

But we have seen instances when one of the sheet is empty - as in it does not even have a header - just the sheet name.
Totally empty. The job is failing as expected although the error message is a little unclear.

`NoSuchElementException: head of empty list`

Is there a way I can identify that the sheet is empty and give out an appropriate error message like `Sheet is Empty` before aborting.
Would be nice to have a clearer message if possible.

Thank you for all the hard work in maintaining and updating spark-excel with new features regularly.
Cheers...

**Comment by nightscape** [comment]
Hi @rbharathkumar, you can use an ordinary `try {} catch` or `scala.util.Try` to catch the exception and turn it into a more meaningful one.

**Reply by rbharathkumar** [reply]
Thank you very much for your response - I will try that approach - is there a specific exception I should be looking for to catch this specific issue ?
try-catch could be raised for any other reason as well right ? Reason I am asking is - in case the error message can be made readable, it will be a lot easier and quicker to react to that failure by contacting relevant source teams... 

Thank you once again for taking the time to respond...

Edit: Just realized that the exception name is there in the error message itself...
So when we encounter `NoSuchElementException` - We can say that the sheet is empty ?
